core/src/main/java/com/yahoo/oak/EntrySet.java

```diff
@@ -190,78 +190,61 @@ int getLastEntryIndex() {
      * Converts external entry-index to internal array index.
      * <p>
      * We use the following terminology:
-     *  - intIdx for the integer's index inside the entries array (referred as a set of integers)
+     *  - longIdx for the long's index inside the entries array (referred as a set of longs)
      *  - entryIdx for the index of entry array (referred as a set of entries)
      *
      * @param entryIdx external entry-index
      * @return the internal array index
      */
-    private static int entryIdx2intIdx(int entryIdx) {
+    private static int entryIdx2LongIdx(int entryIdx) {
         if (entryIdx == 0) { // assumed it is a head pointer access
             return 0;
         }
         return ((entryIdx - 1) * FIELDS) + HEAD_NEXT_INDEX_SIZE;
     }
 
     /**
+<<<<<<< HEAD
+     * getEntryArrayFieldLong atomically reads long field of the entries array.
+     * Could be used with any OFFSET.value
+=======
      * getEntryArrayFieldInt gets the integer field of entry at specified offset for given
      * start of the entry index in the entry array.
      * The field is read atomically as it is a machine word, however the concurrency of the
      * mostly updated value is not ensured as no memory fence is issued.
      */
-    private int getEntryArrayFieldInt(int intFieldIdx, OFFSET offset) {
+    private long getEntryArrayFieldInt(int intFieldIdx, OFFSET offset) {
         return entries[intFieldIdx + offset.value]; // used for NEXT
     }
 
     /**
      * getEntryArrayFieldLong atomically reads two integers field of the entries array.
      * Should be used with OFFSET.VALUE_REFERENCE and OFFSET.KEY_REFERENCE
+>>>>>>> 7ea91e5b1433fe70d89c56c5d0ca6614b323f6ab
      */
     private long getEntryArrayFieldLong(int intStartFieldIdx, OFFSET offset) {
-        assert offset == OFFSET.VALUE_REFERENCE || offset == OFFSET.KEY_REFERENCE;
         long arrayOffset =
-                Unsafe.ARRAY_INT_BASE_OFFSET + (intStartFieldIdx + offset.value) * Unsafe.ARRAY_INT_INDEX_SCALE;
+                Unsafe.ARRAY_LONG_BASE_OFFSET + (intStartFieldIdx + offset.value) * Unsafe.ARRAY_LONG_INDEX_SCALE;
         assert arrayOffset % 8 == 0;
         return UNSAFE.getLong(entries, arrayOffset);
```

Comment:
Why not reading directly from the array? Same comment for `setEntryFieldLong()`.
```java
return entries[intStartFieldIdx + offset.value];
```

Reply:
good point indeed :)